### PR TITLE
Handle "WITH RECURSIVE" abnormal crashes

### PR DESCRIPTION
**Pull request by Leo-XM-Zeng**
This seems to be a PG bug. When a CTE expression with multiple unions is run, this line assertion will necessarily be triggered, causing the program to terminate.

### CODE REVIEW

**Comment by Leo-XM-Zeng** [comment]
#436

**Comment by Leo-XM-Zeng** [comment]
This seems to be a design flaw in pg_get_querydef, which does not adequately account for these scenarios

> This seems to be a PG bug. When a CTE expression with multiple unions is run, this line assertion will necessarily be triggered, causing the program to terminate.

**Comment by Leo-XM-Zeng** [comment]
#435

File: test/regression/sql/cte.sql [hunk 1]
```diff
@@ -6,3 +6,27 @@ WITH modifying_cte AS (
 ) select * from modifying_cte;
 
 DROP TABLE t;
+
+WITH RECURSIVE outermost(x) AS (
```
Comment (JelteF) [inline test/regression/sql/cte.sql:10]: nit: can you move this query before the `DROP TABLE` command. I think it's nice for consistency between tests to keep all the table drops at the end of the sql file.

**Review by JelteF — APPROVED** [review]
LGTM, nice that you also created a patch for the upstream issue in Postgres: https://commitfest.postgresql.org/51/5383/

File: test/regression/sql/cte.sql (hunk 1 above)
Comment (Leo-XM-Zeng) [reply test/regression/sql/cte.sql:8]: Ok, that's done

**Review by JelteF — APPROVED** [review]

**Pull request merged**